src/map/zone_entities.cpp

```diff
@@ -316,6 +369,26 @@ void CZoneEntities::DecreaseZoneCounter(CCharEntity* PChar)
             }
             PChar->PPet = nullptr;
         }
+
+        //remove trust
+        if (PChar->PTrusts.size() > 0)
+        {
+            for (auto& trust : PChar->PTrusts)
+            {
+                trust->PAI->Disengage();
```

Comment:
I'm not as familiar with AI controllers as I'd _like_, but [looking at `ClearTrusts`, it's already calling Despawn()](https://github.com/project-topaz/topaz/blob/release/src/map/entities/charentity.cpp#L490-L499) on the list.